cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -525,3 +536,55 @@ func (c *CStorPoolController) getDeviceIDs(csp *apis.CStorPool) ([]string, error
 	// TODO: Add error handling
 	return pool.GetDeviceIDs(csp)
 }
+
+func (c *CStorPoolController) upgrade(csp *apis.CStorPool) (*apis.CStorPool, error) {
+	if csp.VersionDetails.Current != csp.VersionDetails.Desired {
+		if !isCurrentVersionValid(csp) {
+			return nil, errors.Errorf("invalid current version %s", csp.VersionDetails.Current)
+		}
+		if !isDesiredVersionValid(csp) {
+			return nil, errors.Errorf("invalid desired version %s", csp.VersionDetails.Desired)
+		}
+		// As no other steps are required just change current version to
+		// desired version
+		csp.VersionDetails.Current = csp.VersionDetails.Desired
```

Comment:
Here also, can there be two switch cases.. one for current and another for desired..
Lets bring in that framework